root/static/scripts/common/utility/isDateEmpty.js

```diff
@@ -7,6 +7,17 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
+export function isDateObservableEmpty(
+  date: {
+    day: KnockoutObservable<number | null>,
+    month: KnockoutObservable<number | null>,
+    year: KnockoutObservable<number | null>,
+  },
```

Comment:
Since you check `date == null` below, you might want to allow null values in the type too. (Not that it matters much since the relationship editor isn't Flow-typed.)
```suggestion
  } | null,
```

Reply:
I was actually wondering: can this even be null? It might be that this always has a value.